.circleci/config/base.yml

```diff
@@ -1273,6 +1207,50 @@ commands:
               fi
             fi
 
+  ninja_build_electron:
+    parameters:
+      clean-prebuilt-snapshot:
+        type: boolean
+        default: true
+    steps:
+      - run:
+          name: Electron build
+          no_output_timeout: 60m
+          command: |
+            # On arm platforms we generate a cross-arch ffmpeg that ninja does not seem
+            # to realize is not correct / should be rebuilt.  We delete it here so it is
+            # rebuilt
+            if [ "$TRIGGER_ARM_TEST" == "true" ]; then
+              rm -f src/out/Default/libffmpeg.so
+            fi
+            cd src
+            # Enable if things get really bad
+            # if  [ "$TARGET_ARCH" == "arm64" ] &&[ "`uname`" == "Darwin" ]; then
+            #   diskutil erasevolume HFS+ "xcode_disk" `hdiutil attach -nomount ram://12582912`
+            #   mv /Applications/Xcode-12.beta.5.app /Volumes/xcode_disk/
+            #   ln -s /Volumes/xcode_disk/Xcode-12.beta.5.app /Applications/Xcode-12.beta.5.app
+            # fi
+
+            # Lets generate a snapshot and mksnapshot and then delete all the x-compiled generated files to save space
+            if [ "$USE_PREBUILT_V8_CONTEXT_SNAPSHOT" == "1" ]; then
+              ninja -C out/Default electron:electron_mksnapshot_zip -j $NUMBER_OF_NINJA_PROCESSES
+              ninja -C out/Default tools/v8_context_snapshot -j $NUMBER_OF_NINJA_PROCESSES
+              gn desc out/Default v8:run_mksnapshot_default args > out/Default/mksnapshot_args
+              (cd out/Default; zip mksnapshot.zip mksnapshot_args clang_x64_v8_arm64/gen/v8/embedded.S)
+              if [ "<< parameters.clean-prebuilt-snapshot >>" == "true" ]; then
+                rm -rf out/Default/clang_x64_v8_arm64/gen
+                rm -rf out/Default/clang_x64_v8_arm64/obj
+                rm -rf out/Default/clang_x64_v8_arm64/thinlto-cache
+                rm -rf out/Default/clang_x64/obj
+              fi
+
+              # Regenerate because we just deleted some ninja files 
+              gn gen out/Default --args="import(\"$GN_CONFIG\") import(\"$GN_GOMA_FILE\") $GN_EXTRA_ARGS $GN_BUILDFLAG_ARGS"
+            fi
+            NINJA_SUMMARIZE_BUILD=1 autoninja -C out/Default electron -j $NUMBER_OF_NINJA_PROCESSES
```

Comment:
realise this is just moved code, but doesn't `autoninja` automatically pick `-j`? so we shouldn't need to specify it here.

Reply:
You've asked this before, it also spits out some different output or something iirc, not 100% though, but it stayed in last time